### PR TITLE
plugin-update-4.9.1-4.15.2

### PR DESCRIPTION
**Pull request by igorJarek**

### CODE REVIEW

File: asn1acn.pro [hunk 1]
```diff
@@ -26,10 +26,10 @@
 # Qt Creator linking
 
 IDE_SOURCE_TREE = $$QTC_SOURCE
-isEmpty(IDE_SOURCE_TREE): IDE_SOURCE_TREE = "/opt/qt-creator-dev/qt-creator"
+isEmpty(IDE_SOURCE_TREE): IDE_SOURCE_TREE = "/home/igor/Desktop/4.15.2/qt-creator-4.15.2"
```
Comment (hcorg) [inline asn1acn.pro:29]: well... nope :)

File: src/kitaspectwidget.cpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/****************************************************************************
+**
+** Copyright (C) 2018-2019 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (hcorg) [reply src/kitaspectwidget.cpp:3]: generally - update copyright year in modified files

or update it globally (it's also acceptable)

File: src/kitaspectwidget.h [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/****************************************************************************
+**
+** Copyright (C) 2018-2019 N7 Space sp. z o. o.
```
Comment (hcorg) [inline src/kitaspectwidget.h:3]: year/https

File: src/options-pages/fuzzer.cpp [hunk 4]
```diff
@@ -43,13 +44,13 @@ Fuzzer::Fuzzer(Settings::FuzzerPtr settings)
     setCategory(Constants::SETTINGS_CATEGORY);
 }
 
-bool Fuzzer::matches(const QString &searchKeyWord) const
+bool Fuzzer::matches(const QRegularExpression &regexp) const
 {
     const QStringList keywords{"asn1", "asn.1", "acn", "test", "fuzzer", "asn1scc"};
     for (const auto &keyword : keywords)
-        if (keyword.contains(searchKeyWord, Qt::CaseInsensitive))
+        if (keyword.contains(regexp.pattern(), Qt::CaseInsensitive))
```
Comment (hcorg) [inline src/options-pages/fuzzer.cpp:51]: well... it should be probably the other way around - regexp should be checked for matching with any of the keywords

File: src/options-pages/libraries.cpp [hunk 5]
```diff
@@ -43,13 +43,13 @@ Libraries::Libraries(Settings::LibrariesPtr settings)
     setCategory(Constants::SETTINGS_CATEGORY);
 }
 
-bool Libraries::matches(const QString &searchKeyWord) const
+bool Libraries::matches(const QRegularExpression &regexp) const
 {
     const QStringList keywords{"asn1", "asn.1", "acn", "libraries", "components"};
     for (const auto &keyword : keywords)
-        if (keyword.contains(searchKeyWord, Qt::CaseInsensitive))
+        if (keyword.contains(regexp.pattern(), Qt::CaseInsensitive))
```
Comment (hcorg) [inline src/options-pages/libraries.cpp:50]: should be other way around

File: src/options-pages/service.cpp [hunk 6]
```diff
@@ -47,13 +47,13 @@ Service::Service(Settings::ServicePtr settings)
                     Utils::Icon::Tint));
 }
 
-bool Service::matches(const QString &searchKeyWord) const
+bool Service::matches(const QRegularExpression &regexp) const
 {
     const QStringList keywords{"asn1scc", "daemon", "asn1.exe", "asn1", "asn.1", "acn"};
     for (const auto &keyword : keywords)
-        if (keyword.contains(searchKeyWord, Qt::CaseInsensitive))
+        if (keyword.contains(regexp.pattern(), Qt::CaseInsensitive))
```
Comment (hcorg) [inline src/options-pages/service.cpp:54]: should be other way around

also (as this is a clearly code duplication, when I have to write the same comment 3 times) - try extracting some utility method or check how other plugins do it, maybe there's something available already there

File: templates/wizards/projects/asn1acn/createFilesList.cmake [hunk 7]
```diff
@@ -38,11 +38,12 @@ function(tryCreateFiles fileNames)
 endfunction()
 
 function(appendPersistentListC)
-    set(persistentFiles ${ASN1SCC_C_DIR}/asn1crt.c ${ASN1SCC_C_DIR}/real.c)
```
Comment (hcorg) [inline templates/wizards/projects/asn1acn/createFilesList.cmake:41]: real.c is no longer generated? real/float related functions are included in asn1crt?

File: templates/wizards/projects/asn1acn/createFilesList.cmake [hunk 8]
```diff
@@ -38,11 +38,12 @@ function(tryCreateFiles fileNames)
...
+    set(persistentFiles ${ASN1SCC_C_DIR}/asn1crt.c)
 
     list(FIND ASN1SCC_C_OPTIONS --acn-enc ACN_ENCODING_IDX)
     if (NOT ${ACN_ENCODING_IDX} EQUAL -1)
-        set(persistentFiles ${persistentFiles} ${ASN1SCC_C_DIR}/acn.c)
+        set(persistentFiles ${persistentFiles} ${ASN1SCC_C_DIR}/asn1crt_encoding.c
+            ${ASN1SCC_C_DIR}/asn1crt_encoding_acn.c ${ASN1SCC_C_DIR}/asn1crt_encoding_uper.c)
```
Comment (hcorg) [inline templates/wizards/projects/asn1acn/createFilesList.cmake:46]: asn1crt_encoding - shouldn't that be added always?

why adding UPER encoding, when ACN was selected?

File: templates/wizards/projects/asn1acn/updateSourcesList.pri [hunk 9]
```diff
@@ -105,10 +105,10 @@ names = $$createFileNames($$DISTFILES)
 generateC {
     cNames = $$names
     !isEmpty(names) {
-        cNames += asn1crt real
+        cNames += asn1crt
 
         contains(ASN1SCC_C_OPTIONS, --acn-enc) {
-            cNames += acn
+            cNames += asn1crt_encoding asn1crt_encoding_acn asn1crt_encoding_uper
```
Comment (hcorg) [inline templates/wizards/projects/asn1acn/updateSourcesList.pri:111]: why uper when ACN was selected?

what is in the "encoding" (without any suffixes)? shouldn't it be added "always"?

**Review by hcorg — CHANGES_REQUESTED** [review]

File: src/asn1acn.cpp [hunk 10]
```diff
@@ -170,7 +171,7 @@ bool Asn1AcnPlugin::initialize(const QStringList &arguments, QString *errorStrin
 
     initializeMenus(fuzzerSettings);
 
-    Core::JsExpander::registerQObjectForJs(QLatin1String("Asn1Acn"), new Asn1AcnJsExtension);
+    Core::JsExpander::registerGlobalObject(QLatin1String("Asn1Acn"), []{return new Asn1AcnJsExtension();});
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acn.cpp:174]: done

File: src/asn1acn.cpp [hunk 11]
```diff
@@ -437,24 +438,24 @@ void Asn1AcnPlugin::raiseImportComponentWindow(ProjectExplorer::Project *project
...
     }
 }
 
 #ifdef WITH_TESTS
-QList<QObject *> Asn1AcnPlugin::createTestObjects() const
+QVector<QObject *> Asn1AcnPlugin::createTestObjects() const
 {
-    return QList<QObject *>() << new Libraries::Tests::ModuleMetadataParserTests
+    return QVector<QObject *>() << new Libraries::Tests::ModuleMetadataParserTests
                               << new Libraries::Tests::GeneralMetadataParserTests
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acn.cpp:459]: done

File: src/asn1acnbuildstep.cpp [hunk 12]
```diff
@@ -38,18 +38,23 @@ using namespace Asn1Acn::Internal;
...
 {
     updateEnvironment(buildConfiguration());
-    setOutputParser(new Asn1AcnErrorParser);
     return AbstractProcessStep::init();
 }
 
+void Asn1AcnBuildStep::setupOutputFormatter(Utils::OutputFormatter *formatter)
+{
+    formatter->addLineParser(new Asn1AcnErrorParser());
+    AbstractProcessStep::setupOutputFormatter(formatter);
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acnbuildstep.cpp:55]: I think that it shouldn't, 
please look at :
https://github.com/qt-creator/qt-creator/blob/v4.15.2/src/plugins/qmakeprojectmanager/qmakemakestep.cpp#L203
or
https://github.com/qt-creator/qt-creator/blob/v4.15.2/src/plugins/cmakeprojectmanager/cmakebuildstep.cpp#L292

File: src/kitaspect.cpp [hunk 13]
```diff
@@ -0,0 +1,126 @@
+/****************************************************************************
+**
+** Copyright (C) 2018-2019 N7 Space sp. z o. o.
+** Contact: http://n7space.com
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspect.cpp]: done

File: src/kitaspect.h [hunk 14]
```diff
@@ -0,0 +1,58 @@
+/****************************************************************************
+**
+** Copyright (C) 2018-2019 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspect.h]: done

File: src/kitaspect.h [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+** Contact: http://n7space.com
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspect.h]: done

File: src/kitaspectwidget.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspectwidget.cpp]: done in whole project

File: src/kitaspectwidget.h (hunk 3 above)
Comment (igorJarek) [reply src/kitaspectwidget.h]: done

File: src/kitaspectwidget.cpp [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+#include "kitaspectwidget.h"
+
+#include "kitaspect.h"
+
+using namespace Asn1Acn::Internal;
+
+KitAspectWidget::KitAspectWidget(ProjectExplorer::Kit *kit, const ProjectExplorer::KitAspect *ka)
+    : ProjectExplorer::KitAspectWidget(kit, ka), m_chooser(new Utils::PathChooser)
+{
+    m_chooser->setToolTip("Path to ASN1SCC executable.");
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspectwidget.cpp:35]: done

File: src/asn1acnerrorparser.cpp [hunk 17]
```diff
@@ -32,22 +32,28 @@
...
 using namespace Asn1Acn::Internal;
 
-void Asn1AcnErrorParser::stdError(const QString &line)
+// TODO sprawdzic
+Utils::OutputLineParser::Result Asn1AcnErrorParser::handleLine(const QString &line, Utils::OutputFormat type)
 {
-    if (line.isEmpty())
-        return;
+    if (type == Utils::OutputFormat::StdErrFormat)
+    {
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acnerrorparser.cpp]: done

File: src/asn1acnerrorparser.cpp [hunk 18]
```diff
@@ -32,22 +32,28 @@
 
 using namespace Asn1Acn::Internal;
 
-void Asn1AcnErrorParser::stdError(const QString &line)
+// TODO sprawdzic
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acnerrorparser.cpp:35]: done

File: src/kitaspect.h [hunk 19]
```diff
@@ -0,0 +1,58 @@
...
+class KitAspect : public ProjectExplorer::KitAspect
+{
+public:
+    using Kit = ProjectExplorer::Kit;
+    using Task = ProjectExplorer::Task;
+
+    KitAspect();
+
+    static Utils::Id id();
+    static bool hasAsn1Exe(const Kit *k);
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/kitaspect.h:42]: done

File: templates/wizards/projects/asn1acn/createFilesList.cmake (hunk 7 above)
Comment (igorJarek) [reply templates/wizards/projects/asn1acn/createFilesList.cmake:41]: yes, real.c is no longer generated
no, real/float related functions are included in asn1crt_encoding.h/.c

File: templates/wizards/projects/asn1acn/updateSourcesList.pri (hunk 9 above)
Comment (igorJarek) [reply templates/wizards/projects/asn1acn/updateSourcesList.pri:111]: done

File: asn1acn.pro (hunk 1 above)
Comment (igorJarek) [reply asn1acn.pro:29]: reverted

File: asn1acn.pro [hunk 20]
```diff
@@ -45,6 +45,9 @@ QTC_PLUGIN_RECOMMENDS += \
 
 include($$IDE_SOURCE_TREE/src/qtcreatorplugin.pri)
 
+QMAKE_LFLAGS += -L/home/igor/Desktop/4.15.2/qt-creator-4.15.2/lib/qtcreator/plugins
+QMAKE_LFLAGS += -L/home/igor/Desktop/4.15.2/qt-creator-4.15.2/lib/qtcreator
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply asn1acn.pro:49]: reverted

File: asn1acn.pro [hunk 21]
```diff
@@ -26,10 +26,10 @@
 # Qt Creator linking
 
 IDE_SOURCE_TREE = $$QTC_SOURCE
-isEmpty(IDE_SOURCE_TREE): IDE_SOURCE_TREE = "/opt/qt-creator-dev/qt-creator"
+isEmpty(IDE_SOURCE_TREE): IDE_SOURCE_TREE = "/home/igor/Desktop/4.15.2/qt-creator-4.15.2"
 
 IDE_BUILD_TREE = $$QTC_BUILD
-isEmpty(IDE_BUILD_TREE): IDE_BUILD_TREE = "/opt/qt-creator-dev/build-debug"
+isEmpty(IDE_BUILD_TREE): IDE_BUILD_TREE = "/home/igor/Desktop/4.15.2/build"
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply asn1acn.pro:32]: reverted

File: src/options-pages/fuzzer.cpp (hunk 4 above)
Comment (igorJarek) [reply src/options-pages/fuzzer.cpp:51]: done

File: src/options-pages/libraries.cpp (hunk 5 above)
Comment (igorJarek) [reply src/options-pages/libraries.cpp:50]: done

File: Asn1Acn.json.in [hunk 22]
```diff
@@ -3,10 +3,10 @@
     \"Version\" : \"4.15.2\",
     \"CompatVersion\" : \"4.15.0\",
     \"Vendor\" : \"N7 Space sp. z o.o.\",
-    \"Copyright\" : \"(C) 2017-2022 N7 Space sp. z o.o.\",
+    \"Copyright\" : \"(C) 2022 N7 Space sp. z o.o.\",
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply Asn1Acn.json.in:6]: done

File: asn1acn.pro [hunk 23]
```diff
@@ -1,7 +1,7 @@
 #############################################################################
 ##
-## Copyright (C) 2017-2022 N7 Space sp. z o. o.
-## Contact: http://n7space.com
+## Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply asn1acn.pro]: done

File: src/acndocument.cpp [hunk 24]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/acndocument.cpp]: done

File: src/acndocument.h [hunk 25]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/acndocument.h]: done

File: src/acneditor.cpp [hunk 26]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/acneditor.cpp]: done

File: src/acneditor.h [hunk 27]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/acneditor.h]: done

File: src/asn1acn.cpp [hunk 28]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acn.cpp]: done

File: src/asn1acn.h [hunk 29]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acn.h]: done

File: src/asn1acn_global.h [hunk 30]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2017-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acn_global.h]: done

File: src/asn1acnbuildstep.cpp [hunk 31]
```diff
@@ -1,7 +1,7 @@
 /****************************************************************************
 **
-** Copyright (C) 2018-2019 N7 Space sp. z o. o.
-** Contact: http://n7space.com
+** Copyright (C) 2022 N7 Space sp. z o. o.
```
[the remark this answers is not in the document]
Comment (igorJarek) [reply src/asn1acnbuildstep.cpp]: done

**Review by hcorg — APPROVED** [review]

**Pull request merged**